src/PostgREST/App.hs

```diff
@@ -286,53 +288,78 @@ augumentRequestWithJoin schema allRels request =
   (first formatRelationError . addRelations schema allRels Nothing) request
   >>= addJoinConditions schema
 
-buildReadRequest :: [Relation] -> ApiRequest -> Either Text ReadRequest
-buildReadRequest allRels apiRequest  =
-  augumentRequestWithJoin schema rels =<<
-  first formatParserError (foldr addFilter <$> (foldr addOrder <$> readRequest <*> ords) <*> flts)
+addFiltersOrdersRanges :: ApiRequest -> Either ParseError (ReadRequest -> ReadRequest)
+addFiltersOrdersRanges apiRequest = foldr1 (liftA2 (.)) [
```

Comment:
Unless there is high level function that can make this shorter i think it would have something similar underneath. I say that because i am thinking like:
We are in the context `Either ParserError a` so when we apply a value in that context to a function taking two parameters using `<$>` we get ourselves things like `Either ParserError (ReadRequest -> ReadRequest)`
With no context, we would use simple compose since their types align nicely, but since they are in a context, we need to compose them in that context ... and the only way to do that is applicative (i think), i'll add a comment to make it a bit more clear what's going on and maybe later if some haskell guru looks over it, might suggest something better. My feeling is that if something will be possible, it will be because of the functions in question are folds and not because of the applicative bit